drivers/network/afd/afd/lock.c

```diff
@@ -308,38 +308,44 @@ VOID UnlockBuffers( PAFD_WSABUF Buf, UINT Count, BOOL Address ) {
 /* Produce a kernel-land handle array with handles replaced by object
  * pointers.  This will allow the system to do proper alerting */
 PAFD_HANDLE LockHandles( PAFD_HANDLE HandleArray, UINT HandleCount ) {
```

Comment:
I have a problem with this function (and its caller in the AFD driver): There doesn't seem to be any sort of validation of the HandleCount value, with respect to the reported buffer size passed to the IOCTL handler, and thus no guard that the loop for copying the handles won't do a buffer overread.